app/src/main/java/lt/vilnius/tvarkau/MainActivity.java

```diff
@@ -16,6 +29,14 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.home);
 
         ButterKnife.bind(this);
+
+        handleSignInButton();
+    }
+
+    private void handleSignInButton() {
+        if(!SharedPrefsManager.instance(this).getIsUserAnonymous()){
```

Comment:
Good point noticed by @vycius - use application context instead of this activity.